### PR TITLE
Add more details in the status about ordered and unordered

### LINKED ISSUE #732: Add DeliveryOrder status condition or annotation for Trigger
**Problem**
In https://github.com/knative-sandbox/eventing-kafka-broker/pull/701 we introduced a new annotation to configure the delivery order and during the discussion there, we come up to the conclusion that would like to have a status condition to give users feedback on it since we don't have a way to signal that the annotation has been read correctly.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
- Event consumer

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1

**Additional context (optional)**

- Discussion in https://github.com/knative-sandbox/eventing-kafka-broker/pull/701

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fixes #732

## Proposed Changes

- :gift: Add some details in the existing subscriber resolved condition about the delivery order. I didn't added a new condition for it because it's quite complicated and requires several changes in eventing too

**Release Note**

```release-note
- :gift: Add some details in the existing subscriber resolved condition about the delivery order.
```

**Docs**

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
cc @pierDipi

File: control-plane/pkg/reconciler/trigger/trigger.go [hunk 1]
```diff
@@ -142,7 +142,9 @@ func (r *Reconciler) reconcileKind(ctx context.Context, trigger *eventing.Trigge
 	if err != nil {
 		return statusConditionManager.failedToResolveTriggerConfig(err)
 	}
-	statusConditionManager.subscriberResolved()
+	statusConditionManager.subscriberResolved(
+		fmt.Sprintf("Subscriber will receive events with the delivery order: %s", triggerConfig.DeliveryOrder.String()),
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/trigger.go:146]: A bit more concise?
```suggestion
		fmt.Sprintf("Delivery order: %s", triggerConfig.DeliveryOrder.String()),
```

File: control-plane/pkg/reconciler/trigger/trigger_lifecycle.go [hunk 2]
```diff
@@ -122,6 +122,10 @@ func (m *statusConditionManager) failedToUpdateDispatcherPodsAnnotation(err erro
 	)
 }
 
-func (m *statusConditionManager) subscriberResolved() {
-	m.Trigger.Status.MarkSubscriberResolvedSucceeded()
+func (m *statusConditionManager) subscriberResolved(details string) {
+	m.Trigger.GetConditionSet().Manage(&m.Trigger.Status).MarkTrueWithReason(
+		eventing.TriggerConditionSubscriberResolved,
+		"SubscriberResolved",
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/trigger_lifecycle.go:128]: Any public constant?

File: control-plane/pkg/reconciler/trigger/trigger.go (hunk 1 above)
Comment (slinkydeveloper) [reply control-plane/pkg/reconciler/trigger/trigger.go:146]: I prefer this wording, because the name of the reason is still `SubscriberResolved`...

Comment (pierDipi) [reply control-plane/pkg/reconciler/trigger/trigger.go:146]: Yeah, make sense, thanks!

**Review by pierDipi — APPROVED** [review]

**Pull request merged**